oci/__init__.py

```diff
@@ -490,6 +826,77 @@ def replicate_blob(blob: om.OciBlobRef) -> om.OciBlobRef:
     )
 
 
+async def replicate_blobs_async(
+    src_ref: str,
+    src_oci_manifest: om.OciImageManifest,
+    tgt_ref: str,
+    oci_client: oc.AsyncClient,
+    blob_overwrites: dict[om.OciBlobRef, bytes | io.BytesIO],
+) -> om.OciImageManifest:
+    '''
+    replicates blobs from given oci-image-ref to the specified target-ref, optionally replacing
+    the specified blobs. This is particularly useful for replacing some "special" blobs, such
+    as a component-descriptor layer blob or the config-blob.
+
+    Note that the uploaded artifact must be finalised after the upload by a "manifest-put".
+    '''
+    blob_overwrites = {k.digest:v for k,v in blob_overwrites.items()}
+
+    async def replicate_blob_async(blob: om.OciBlobRef) -> om.OciBlobRef:
+        if blob_overwrite_bytes := blob_overwrites.get(blob.digest):
+            logger.info(f'Replicate with overwriting {blob=}')
+
+            if hasattr(blob_overwrite_bytes, 'read'):
```

Comment:
hmmm...... does it actually make sense to accept a (synchronous) reader/filelike-object in async-flavour of this function?